containers/kubernetes/how-to/manage-kubeconfig-with-iam.mdx

```diff
@@ -0,0 +1,71 @@
+---
+meta:
+  title: How to manage Kubeconfig files with IAM
+  description: This page explains how to manage Kubeconfig files with IAM
+content:
+  h1: How to manage Kubeconfig files with IAM
+  paragraph: This page explains how to manage Kubeconfig files with IAM
+tags: kubernetes kapsule-cluser
+dates:
+  validation: 2024-01-09
+  posted: 2024-01-09
+categories:
+  - kubernetes
+---
+
+
+A Kubeconfig file serves as a repository for essential credentials required to access a Kubernetes cluster and manage deployments.  
+This file contains critical information, including cluster details, authentication data, and context information, allowing users to oversee and configure their Kubernetes clusters effectively.
+
+Scaleway's Identity and Access Management (IAM) is now being used to generate Kubeconfig files. This integration offers several advantages:
+
+- **Improved user administration:**
+   - When removing a cluster user, there is no longer a need to reset the admin token, mitigating security risks.
+- **Increased Cluster Security:**
+   - Each user possesses a unique credential, enhancing security.
+   - Audit logs facilitate the identification of specific Kubernetes users.
+- **Enhanced Application Security:**
+   - Applications within the cluster are authenticated using specific tokens.
+   - This authentication method is particularly useful when a pod needs to interact with the Kubernetes API, as it can authenticate using its associated Service Account.
+
+[Identity and Access Management (IAM)](https://www.scaleway.com/en/docs/identity-and-access-management/iam/concepts/#iam) provides control over resource access. IAM policies enable the configuration of permissions for Kubernetes Kapsule clusters at the Project level.
+
+An [IAM policy](https://www.scaleway.com/en/docs/identity-and-access-management/iam/concepts/#policy) defines the permissions for users, groups, and applications within an Organization. It consists of a [principal](https://www.scaleway.com/en/docs/identity-and-access-management/iam/concepts/#principal) (the user, group, or application to which it applies) and IAM rules that specify permission sets and their scope.
+
+The [combination of IAM and Kubernetes RBAC](/containers/kubernetes/reference-content/set-iam-permissions-and-implement-rbac/) allows to define finely granulated access levels for cluster users.
+
+<Message type="important">
+    Entering a deprecation cycle, the former authentication system (using static admin tokens) remains active until Q4/2024: all your legacy Kubeconfig files keep access to their clusters for the time being. For security measure you can still make use of the "reset admin token" feature !!!ADD LINK HERE!!!, either through our Kubernetes API or Scaleway’s CLI.
```

Comment:
Just a reminder to add the !!!LINK HERE!!! hehe

Reply:
Thanks. Added the link :)